homeassistant/components/nasweb/coordinator.py

```diff
@@ -0,0 +1,199 @@
+        _LOGGER.debug("Updated %s data", self.name)
+        if self._listeners:
+            self._schedule_last_update_check()
+        self.async_update_listeners()
+
+    @callback
+    def async_update_listeners(self) -> None:
+        """Update all registered listeners."""
+        for update_callback, _ in list(self._listeners.values()):
+            update_callback()
+
+    async def _handle_max_update_interval(self, now: datetime) -> None:
+        """Handle max update interval occurrence."""
+        self._unsub_last_update_check = None
+        if self._listeners:
+            self.async_update_listeners()
+
+    def _schedule_last_update_check(self) -> None:
+        self._async_unsub_last_update_check()
+        now = self._hass.loop.time()
+        next_check = (
+            now + timedelta(seconds=STATUS_UPDATE_MAX_TIME_INTERVAL).total_seconds()
+        )
+        self._unsub_last_update_check = event.async_call_at(
+            self._hass,
+            self._job,
+            next_check,
+        )
+
+    def _async_unsub_last_update_check(self) -> None:
+        """Cancel any scheduled update check call."""
+        if self._unsub_last_update_check:
+            self._unsub_last_update_check()
+            self._unsub_last_update_check = None
+
+    async def handle_push_notification(self, notification: dict) -> None:
+        """Handle incoming push notification from NASweb."""
+        msg_type = notification.get(KEY_TYPE)
+        _LOGGER.debug("Received push notification: %s", msg_type)
+
+        if msg_type == TYPE_STATUS_UPDATE:
+            await self.process_status_update(notification)
+            self._last_update = time.time()
+
+    async def process_status_update(self, new_status: dict) -> None:
+        """Process status update from NASweb."""
+        new_objects = self.webio_api.update_device_status(new_status)
+        new_outputs = new_objects[KEY_OUTPUTS]
+        if len(new_outputs) > 0:
+            await self._add_switch_entities(new_outputs)
+        self.async_set_updated_data(self.data)
+
+    async def _add_switch_entities(self, switches: list[NASwebOutput]) -> None:
+        # pylint: disable=import-outside-toplevel
+        from .switch import RelaySwitch
+
+        new_switch_entities: list[RelaySwitch] = []
+        for nasweb_output in switches:
+            if not isinstance(nasweb_output, NASwebOutput):
+                _LOGGER.error("Cannot create RelaySwitch without NASwebOutput")
+                continue
+            relay_switch = RelaySwitch(self, nasweb_output)
+            new_switch_entities.append(relay_switch)
+        integration_platforms = async_get_platforms(self._hass, DOMAIN)
+        switch_platform = None
+        for p in integration_platforms:
+            if p.domain == Platform.SWITCH:
+                switch_platform = p
+                break
+        if switch_platform is not None:
+            await switch_platform.async_add_entities(new_switch_entities)
```

Comment:
Move this to `switch.py`:
in `switch.py::async_setup_entry` you add a coordinator listener, which adds switches which have not been seen before. Then you call that listener once to create the switches which are already known.

Here's an example https://github.com/home-assistant/core/blob/95b532df41a7139c8f3b24149ae3f07d3a695fdf/homeassistant/components/airgradient/button.py#L44-L83